libs/auth/react/src/lib/Authenticator/CheckIdpSession.tsx

```diff
@@ -18,45 +20,42 @@ export interface SessionInfo {
   isExpired?: boolean
 }
 
-const messageEventName = 'message'
-
-const getReturnUrl = (history: History, { redirectPath }: AuthSettings) => {
-  const returnUrl = history.location.pathname + history.location.search
-  if (redirectPath && returnUrl.startsWith(redirectPath)) {
-    return '/'
-  }
-  return returnUrl
+interface SessionInfoMessage {
+  type: 'SessionInfo'
 }
 
+const messageEventName = 'message'
+
 export const CheckIdpSession = () => {
-  const history = useHistory()
   const reducerInstance = useReducer(reducer, initialState)
   const [state, dispatch] = reducerInstance
   const userManager = getUserManager()
   const authSettings = getAuthSettings()
   const iframeSrc = `${authSettings.authority}${authSettings.checkSessionPath}`
   const [iframeChecksum, setIframeChecksum] = useState(0)
+  const [sessionTimeout, setSessionTimeout] = useState<ReturnType<
+    typeof setTimeout
+  > | null>(null)
 
-  let sessionTimeout: ReturnType<typeof setTimeout> | null = null
-
-  const signIn = useCallback(
-    async function signIn() {
+  const signOut = useCallback(
+    async function signOut() {
       dispatch({
-        type: ActionType.SIGNIN_START,
+        type: ActionType.LOGGING_OUT,
       })
-      return userManager.signinRedirect({
-        state: getReturnUrl(history, authSettings),
+      await userManager.signoutRedirect({
+        post_logout_redirect_uri: authSettings.baseUrl,
```

Comment:
Can we somehow clear the user session and refresh the page? I.e. instead of triggering signoutRedirect? I prefer coming back to where I was...

Reply:
Note that the baseUrl may not currently be a valid post redirect uri. The normal signout takes the user to the island.is front page.

Reply:
Fixed :)